src/WebJobs.Script.Grpc/MessageExtensions/GrpcMessageConversionExtensions.cs

```diff
@@ -88,25 +85,23 @@ public static async Task<TypedData> ToRpc(this object value, ILogger logger, Grp
         internal static TypedData ToRpcCollection(this object value)
         {
             TypedData typedData;
-            if (value is byte[][] arrBytes)
-            {
-                typedData = arrBytes.ToRpcByteArray();
-            }
-            else if (value is string[] arrStr)
-            {
-                typedData = arrStr.ToRpcStringArray();
-            }
-            else if (value is double[] arrDouble)
-            {
-                typedData = arrDouble.ToRpcDoubleArray();
-            }
-            else if (value is long[] arrLong)
-            {
-                typedData = arrLong.ToRpcLongArray();
-            }
-            else
-            {
-                typedData = value.ToRpcDefault();
+            switch (value)
+            {
+                case byte[][] arrBytes:
+                    typedData = arrBytes.ToRpcByteArray();
+                    break;
+                case string[] arrStr:
+                    typedData = arrStr.ToRpcStringArray();
+                    break;
+                case double[] arrDouble:
+                    typedData = arrDouble.ToRpcDoubleArray();
+                    break;
+                case long[] arrLong:
+                    typedData = arrLong.ToRpcLongArray();
+                    break;
+                default:
+                    typedData = value.ToRpcDefault();
+                    break;
```

Comment:
I'm not sure if there's a reason we're pinned at C# 7.2, but if the team is interested things like this and the above get a bit more terse in C# 8 and above, for example this could be:
```cs
internal static TypedData ToRpcCollection(this object value)
{
    return value switch
    {
        byte[][] arrBytes => arrBytes.ToRpcByteArray(),
        string[] arrStr => arrStr.ToRpcStringArray(),
        double[] arrDouble => arrDouble.ToRpcDoubleArray(),
        long[] arrLong => arrLong.ToRpcLongArray(),
        _ => value.ToRpcDefault(),
    };
}
```

This could just be preference as well that I'm just learning - wanted to point it out as I see some things new language features can make easier though!

Reply:
I like how simple that new approach is! @brettsam & @fabiocav might be able to answer regarding being pinned to C# 7.2?

Reply:
@NickCraver turns out we're not actually pinned and use latest: https://github.com/Azure/azure-functions-host/commit/a68438b21a65ca7d5c209d4e23bad9f52276ffda

Something is off with common.props in my PR - will fix that then start using this new switch approach :D

Reply:
Ahhhh, that's what was throwing me - this is baseline behind dev, awesome - good catch!

Reply:
Thanks to @brettsam ofc :)